Assets/Content/Systems/Interactions/FlashlightInteraction.cs

```diff
@@ -14,6 +14,14 @@ public bool ClientStart(InteractionEvent interactionEvent)
                 if (interactionEvent.Target is Flashlight flashlight)
                 {
                     flashlight.light.enabled = !flashlight.light.enabled;
+                    if (flashlight.light.enabled)
+                    {
+                        flashlight.bulb.GetComponent<MeshRenderer>().sharedMaterial = flashlight.onmat;
+                    }
+                    else
+                    {
+                        flashlight.bulb.GetComponent<MeshRenderer>().sharedMaterial = flashlight.offmat;
+                    }
```

Comment:
Instead of using that many references, create a _bool_ "on" or "enabled". And get that **Mesh Renderer** once.

`if (interactionEvent.Target is Flashlight flaslight) 
{
        on = !on;
        Material lightMaterial = flashlight.bulb.GetComponent<MeshRenderer>().sharedMaterial;
        lightMaterial = on ? flashlight.materialOff : flashlight.materialOn;
}`

Reply:
Im just gonna yoink that because it's a good way of doing it.